tests/integration-service/namespace-backed-environments.go

```diff
@@ -87,11 +88,10 @@ var _ = framework.IntegrationServiceSuiteDescribe("Namespace-backed Environment
 		})
 
 		It("triggers a build PipelineRun", Label("integration-service"), func() {
-			pipelineRun, err = f.AsKubeDeveloper.IntegrationController.GetBuildPipelineRun(componentName, applicationName, testNamespace, false, "")
 			Expect(f.AsKubeDeveloper.HasController.WaitForComponentPipelineToBeFinished(originalComponent, "",
 				f.AsKubeAdmin.TektonController, &has.RetryOptions{Retries: 2, Always: true})).To(Succeed())
-
-			Expect(pipelineRun.Annotations[snapshotAnnotation]).To(Equal(""))
```

Comment:
Curious, why was this removed? we need it to ensure that the snapshot annotation on build PLRs is initially empty, and then [later gets populated](https://github.com/redhat-appstudio/e2e-tests/pull/995/files#diff-41ef80bee886bb589e7d08af7a466e625c230cd50821a529ffcd5f8d284bf5b4R108) with the name of the snapshot right after creation.

Everything else LGTM!

Reply:
I deleted it because, as the code is now, we are getting the pipelineRun after `WaitForComponentPipelineToBeFinished`. So, the snapshotAnnotation will be not empty.

And because if we verify the snapshot is empty before `WaitForComponentPipelineToBeFinished`, we will just check that is empty in the first attempt. If there is a need to retry, we are not checking. Sorry if I am understanding wrong.

Reply:
I will add it before and after to ensure that it is working as expected. Thank you so much for your insights!

Reply:
That's a fair point, I think you're understanding is correct. Thanks!